pkg/controller/registry/resolver/cache/predicates.go

```diff
@@ -329,3 +330,41 @@ func (c countingPredicate) String() string {
 func CountingPredicate(p Predicate, n *int) Predicate {
 	return countingPredicate{p: p, n: n}
 }
+
+type evaluatorPredicate struct {
+	evaluator constraints.Evaluator
+	rule      string
+	message   string
+}
+
+func (ep *evaluatorPredicate) Test(entry *Entry) bool {
+	props := make([]map[string]interface{}, len(entry.Properties))
+	for i, p := range entry.Properties {
+		var v interface{}
+		if err := json.Unmarshal([]byte(p.Value), &v); err != nil {
+			continue
```

Comment:
should we at least log that there was an unmarshalling error? Or, maybe panic?

Reply:
This is `continue` behavior is consistent with other predicates with the intention to ignore bad properties. Usually, bad properties shouldn't even make it to the cluster as bundle validation should catch this. A panic would cause the entire resolver action to fail which essentially means one bad property in one bundle will cause OLM to stop working entirely. I can go with `panic` for now given there is no logging here.

Reply:
Yeah, that makes sense. Silent errors aren't the best tho. Is it very difficult to have attach a logger to the class? Maybe just instantiate a global one for the package? I'm not sure how logging works in Go and our projects yet. If it's too much effort or too outside the scope of the PR and panic can be disruptive to the user, I'd suggest creating a tracking issue and link it in a comment.

Reply:
I think logging the error is the better solution here but as you said, it is not in-scope of this PR to rework the predicate to include logging. The `panic` solution is more disruptive unfortunately. It is kind of lose-lose situation. At best, we can keep it status quo with the `continue` hence my original code. Anyway, let me see what I can do here.